nautobot/dcim/tests/test_api.py

```diff
@@ -3488,6 +3488,21 @@ def test_patching_primary_ip_success(self):
             vdc.refresh_from_db()
             self.assertEqual(vdc.primary_ip6, ip_v6)
 
+    def test_changing_device_on_vdc_raise_validation_error(self):
```

Comment:
Similar test needed in `test_views.py`?

Also, should we add a check in the model `clean()` itself?

Reply:
Hmmm, how do I compare the previous `device` to the new `device` assigned in model `clean()`, any hints?

Reply:
Also, I am not sure if a test is needed in test_views.py. I am just disabling the `device` field and marking it as `non-required`. No validation error should be raised.

Reply:
https://github.com/nautobot/nautobot/blob/develop/nautobot/ipam/models.py#L1142-L1146